src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -61,202 +66,339 @@ static ColliderCylinderInit D_80A71EC0 = {
     { 20, 46, 0, { 0, 0, 0 } },
 };
 
-static CollisionCheckInfoInit2 D_80A71EEC = { 0x00, 0x0000, 0x0000, 0x0000, 0xFF };
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
 
-static UNK_PTR D_80A71EF8[] = { 0x060005C8, 0x06000DC8, 0x060015C8, NULL };
-static UNK_PTR D_80A71F08[] = { 0x0600057C, 0x0600067C, 0x0600077C, NULL };
-static UNK_PTR D_80A71F18[] = { 0x060004C8, NULL };
-static UNK_PTR D_80A71F20[] = { 0x060005FC, 0x060009FC, 0x06000DFC, NULL };
-static UNK_PTR D_80A71F30[] = { 0x060005FC, 0x060006FC, 0x060007FC, NULL };
-static UNK_PTR D_80A71F40[] = { 0x060007C8, 0x06000FC8, 0x060017C8, NULL };
+// NULL-terminated arrays of eye textures
+static void* sEyeTexturesAOB[] = { 0x060005C8, 0x06000DC8, 0x060015C8, NULL };
+static void* sEyeTexturesAHG7[] = { 0x0600057C, 0x0600067C, 0x0600077C, NULL };
+static void* sEyeTexturesBBA[] = { 0x060004C8, NULL };
+static void* sEyeTexturesBJI13[] = { 0x060005FC, 0x060009FC, 0x06000DFC, NULL };
+static void* sEyeTexturesBOJ2[] = { 0x060005FC, 0x060006FC, 0x060007FC, NULL };
+static void* sEyeTexturesBOB[] = { 0x060007C8, 0x06000FC8, 0x060017C8, NULL };
 
 typedef struct {
     /* 0x00 */ s16 objectId;
-    /* 0x04 */ Gfx* dList;
-    /* 0x08 */ UNK_PTR* unk_8;
-} EnHyUnknownStruct1; // size = 0xC
-
-static EnHyUnknownStruct1 D_80A71F50[] = {
-    { OBJECT_AOB, gDogLadyHeadDL, D_80A71EF8 }, { OBJECT_BOB, 0x06003B78, D_80A71F40 },
-    { OBJECT_BOJ, 0x060026F0, D_80A71F30 },     { OBJECT_BOJ, 0x060052E0, NULL },
-    { OBJECT_BOJ, 0x06005528, NULL },           { OBJECT_BOJ, 0x06005738, NULL },
-    { OBJECT_BOJ, 0x060059B0, NULL },           { OBJECT_AHG, 0x060030F0, D_80A71F08 },
-    { OBJECT_AHG, 0x06005508, NULL },           { OBJECT_AHG, 0x06005728, NULL },
-    { OBJECT_BBA, 0x06002948, D_80A71F18 },     { OBJECT_CNE, 0x06001300, NULL },
-    { OBJECT_CNE, 0x06002860, NULL },           { OBJECT_BJI, 0x06002560, D_80A71F20 },
-    { OBJECT_BJI, 0x06003F68, NULL },           { OBJECT_COB, 0x06001300, NULL },
+    /* 0x04 */ Gfx* headDList;
+    /* 0x08 */ void** eyeTextures;
+} EnHyHeadInfo; // size = 0xC
+
+typedef enum {
+    /*  0 */ ENHY_HEAD_AOB,
+    /*  1 */ ENHY_HEAD_BOB,
+    /*  2 */ ENHY_HEAD_BOJ_2,
+    /*  3 */ ENHY_HEAD_BOJ_3,
+    /*  4 */ ENHY_HEAD_BOJ_4,
+    /*  5 */ ENHY_HEAD_BOJ_5,
+    /*  6 */ ENHY_HEAD_BOJ_6,
+    /*  7 */ ENHY_HEAD_AHG_7,
+    /*  8 */ ENHY_HEAD_AHG_8,
+    /*  9 */ ENHY_HEAD_AHG_9,
+    /* 10 */ ENHY_HEAD_BBA,
+    /* 11 */ ENHY_HEAD_CNE_11,
+    /* 12 */ ENHY_HEAD_CNE_12,
+    /* 13 */ ENHY_HEAD_BJI_13,
+    /* 14 */ ENHY_HEAD_BJI_14,
+    /* 15 */ ENHY_HEAD_COB
+} EnHyHeadIndex;
+
+// indexed by EnHyHeadIndex
+static EnHyHeadInfo sHeadInfo[] = {
+    /*  0 */ { OBJECT_AOB, gDogLadyHeadDL, sEyeTexturesAOB },
```

Comment:
I think the comments in all these arrays should use the enum values since the numbers themselves don't mean anything. And if we use the enums then the "indexed by" comments are probably not necessary either.